art/defences/trainer/adversarial_trainer_oaat_pytorch.py

```diff
@@ -0,0 +1,1224 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2023
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This is a PyTorch implementation of the Oracle Aligned Adversarial Training (OAAT) protocol
+for adversarial training for defence against larger perturbations.
+
+| Paper link: https://link.springer.com/chapter/10.1007/978-3-031-20065-6_18
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+import time
+from typing import Optional, Tuple, TYPE_CHECKING, List, Dict, Union
+from collections import OrderedDict
+import six
+
+import numpy as np
+from tqdm.auto import trange
+from art import config
+
+from art.defences.trainer.adversarial_trainer_oaat import AdversarialTrainerOAAT
+from art.estimators.classification.pytorch import PyTorchClassifier
+from art.data_generators import DataGenerator
+from art.attacks.attack import EvasionAttack
+from art.utils import check_and_transform_label_format, random_sphere
+from art.config import ART_NUMPY_DTYPE
```

Comment:
Order imports according to PEP
```suggestion
from collections import OrderedDict
import logging
import time
from typing import Optional, Tuple, TYPE_CHECKING, List, Dict, Union

import six
import numpy as np
from tqdm.auto import trange

from art import config
from art.defences.trainer.adversarial_trainer_oaat import AdversarialTrainerOAAT
from art.estimators.classification.pytorch import PyTorchClassifier
from art.data_generators import DataGenerator
from art.attacks.attack import EvasionAttack
from art.utils import check_and_transform_label_format, random_sphere
from art.config import ART_NUMPY_DTYPE
```

Reply:
changed the order